### ISSUE TITLE
Passing the final loss to arbitrary vjps in the computation graph.

### ISSUE THREAD

**Issue by freckletonj** [HIPS/autograd#498]
*TL;DR*: My missing piece is being able to pass the final `loss` to arbitrary `vjp`s in the computation graph.

There's a really cool technique called Feedback Alignment, and it's super simple.

[Random Feedback Weights Support Learning in Deep Neural Networks](https://arxiv.org/abs/1411.0247)

[Direct Feedback Alignment Provides Learning in Deep Neural Networks](https://arxiv.org/pdf/1609.01596.pdf)

Basically, after projecting a tensor through a weight matrix (`np.dot(x, W)`), don't backprop the loss wrt `W`, make a second _static_, unchanging matrix `B` and project the loss backward through that. Gradients will not check out, but it will learn at the same speed as if you used `W.T`.

The benefits include making sure gradients can't explode/vanish.

I'm still new to `autograd`, so, still poking about but here's an example of vanilla Feedback Alignment.

```python
@primitive
def dot_fa(x, W, B):
    return np.dot(x, W)

defvjp(dot_fa,
       lambda ans, x, W, B: lambda g: np.dot(g, B), # B.shape == W.T.shape
       lambda ans, x, W, B: lambda g: np.outer(x, g),
       None)
```

AFAIK, you can use that in place of `np.dot` any time you like, works great in RNNs where gradients are more likely to explode. I've tested it in a simple FFNN that learns MNIST just as fast as vanilla backprop.

file:///home/josh/Pictures/0-screenshots/window_2019-05-01-17-00-31.png

There's a Direct and an Indirect version of that FA technique, but requires the `vjp` to know what the final loss was. Is there a way to pass that back in at other stages.

Specifically, all I need to calculate the `vjp` is to do `np.dot(g, B)` where `g` is the gradient of loss at the final output, not the current calculation.
